colcon_core/dependency_descriptor.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2016-2018 Dirk Thomas
+# Licensed under the Apache License, Version 2.0
+
+
+class DependencyDescriptor:
+    """A descriptor for a package dependency."""
+
```

Comment:
Please use `__slots__` to limit the members to a known set (`name`, `metadata`).